algebra/matrix.c

```diff
@@ -367,36 +367,57 @@ int matrix_cmp(matrix_t *A, matrix_t *B)
 
 	if (A->transposed) {
 		if (B->transposed) {
+			if (A->cols != B->cols || A->rows != B->rows) {
```

Comment:
Not in scope of this PR, but please add comments as in `matrix_prod()`:
```suggestion
			/* both transposed */
			if (A->cols != B->cols || A->rows != B->rows) {
```

Reply:
Still no comments :)